src/utils/water-network-data.js

```diff
@@ -78,15 +78,21 @@ export const displayWaterNetworkFeaturesInMapViewport = async (map) => {
   const box = bboxPolygon(mapBounds);
   map.getSource("bbox").setData(box);
 
-  await getFeaturesInBoundingBox("HydroNode", mapBounds).then((hydroNodes) => {
-    map.getSource("hydroNodes").setData(hydroNodes);
-  });
+  await getFeaturesInBoundingBox("HydroNode", mapBounds)
+    .then((hydroNodes) => {
+      map.getSource("hydroNodes").setData(hydroNodes);
+    })
+    .catch((error) => {
+      console.error(error);
+    });
```

Comment:
Not sure if this is the best way to catch and handle the error when there's no hydro nodes in the bbox

but found that the 404 response got in the way of rendering watercourse links that are 100% there (because we're looking at water 😂) but this error stops them from being rendered:

before:
![image](https://user-images.githubusercontent.com/32230328/197270116-78a83998-cfb0-4b52-9261-0ea984650243.png)

after:
![image](https://user-images.githubusercontent.com/32230328/197270133-c72d057d-b255-4c95-aef0-3b14bff17920.png)

Reply:
yeah I see what you're saying.. this is fine I think. Might be worth leaving a comment about why we're allowing this particular error to pass through, and maybe checking that we only allow 404s through and not other errors? Or maybe we could make use of [Promise.all](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all)? We'd have to check the result array but we could handle that.. if all requests were either success or 404 we'd show the results, if any actually failed we can error